eth_abi deprecation warning for abi.encode_abi

* Version: 6.0.0b4
* Python: 3.10
* OS: osx

### What was wrong?

Contract calls raise a `DeprecationWarning` due to `encode_abi` being renamed to `encode` in `eth_abi`:

```
    return self.contract.functions.decimals().call()
[snip]/lib/python3.10/site-packages/web3/contract.py:1228: in call
    return call_contract_function(
[snip]/lib/python3.10/site-packages/web3/contract.py:1982: in call_contract_function
    call_transaction = prepare_transaction(
[snip]/lib/python3.10/site-packages/web3/_utils/contracts.py:263: in prepare_transaction
    prepared_transaction["data"] = encode_transaction_data(
[snip]/lib/python3.10/site-packages/web3/_utils/contracts.py:303: in encode_transaction_data
    return add_0x_prefix(encode_abi(w3, fn_abi, fn_arguments, fn_selector))
[snip]/lib/python3.10/site-packages/web3/_utils/contracts.py:218: in encode_abi
    encoded_arguments = w3.codec.encode_abi(
-------
    def encode_abi(self, types: Iterable[TypeStr], args: Iterable[Any]) -> bytes:
        """
        Encodes the python values in ``args`` as a sequence of binary values of
        the ABI types in ``types`` via the head-tail mechanism.

        :param types: An iterable of string representations of the ABI types
            that will be used for encoding e.g.  ``('uint256', 'bytes[]',
            '(int,int)')``
        :param args: An iterable of python values to be encoded.

        :returns: The head-tail encoded binary representation of the python
            values in ``args`` as values of the ABI types in ``types``.
        """
>       warnings.warn(
            "abi.encode_abi() and abi.encode_abi_packed() are deprecated and will be "
            "removed in version 4.0.0 in favor of abi.encode() and "
            "abi.encode_packed(), respectively",
            category=DeprecationWarning,
        )
E       DeprecationWarning: abi.encode_abi() and abi.encode_abi_packed() are deprecated and will be removed in version 4.0.0 in favor of abi.encode() and abi.encode_packed(), respectively
```


### How can it be fixed?

Call `encode` instead of `encode_abi` as directed by the `eth_abi` warning.

---
**Note:** We prefer to use issues to track our work. If you think you've encountered a bug in web3py or 
have a feature request, you're in the right place. If you have implementation or usage questions, 
please refer to our [documentation](https://web3py.readthedocs.io/en/latest/) and/or join the conversation 
on [discord](https://discord.gg/GHryRvPB84).

--- a/newsfragments/2621.feature.rst
+++ b/newsfragments/2621.feature.rst
@@ -1,0 +1,2 @@
+update all references to deprecated `eth_abi.encode_abi` to `eth_abi.encode`
+

--- a/tests/core/contracts/test_contract_constructor_encoding.py
+++ b/tests/core/contracts/test_contract_constructor_encoding.py
@@ -67,7 +67,7 @@ def test_contract_constructor_encoding_encoding(
     )
     encoded_args = "0x00000000000000000000000000000000000000000000000000000000000004d26162636400000000000000000000000000000000000000000000000000000000"  # noqa: E501
     expected_ending = encode_hex(
-        w3.codec.encode_abi(["uint256", "bytes32"], [1234, b"abcd"])
+        w3.codec.encode(["uint256", "bytes32"], [1234, b"abcd"])
     )
     assert expected_ending == encoded_args
     assert deploy_data.endswith(remove_0x_prefix(expected_ending))
@@ -86,7 +86,7 @@ def test_contract_constructor_encoding_encoding_warning(
         encoded_args = "0x00000000000000000000000000000000000000000000000000000000000004d26162636400000000000000000000000000000000000000000000000000000000"  # noqa: E501
 
         expected_ending = encode_hex(
-            w3.codec.encode_abi(["uint256", "bytes32"], [1234, b"abcd"])
+            w3.codec.encode(["uint256", "bytes32"], [1234, b"abcd"])
         )
         assert expected_ending == encoded_args
         assert deploy_data.endswith(remove_0x_prefix(expected_ending))
@@ -114,7 +114,7 @@ def test_contract_constructor_encoding_encoding_strict(
     )
 
     expected_ending = encode_hex(
-        w3_strict_abi.codec.encode_abi(["uint256", "bytes32"], [1234, bytes_arg])
+        w3_strict_abi.codec.encode(["uint256", "bytes32"], [1234, bytes_arg])
     )
     assert expected_ending == encoded_args
     assert deploy_data.endswith(remove_0x_prefix(expected_ending))

--- a/tests/core/contracts/test_extracting_event_data.py
+++ b/tests/core/contracts/test_extracting_event_data.py
@@ -337,7 +337,7 @@ def test_argument_extraction_strict_bytes_types(
     event_topic = emitter_log_topics.LogListArgs
     assert event_topic in log_entry["topics"]
 
-    encoded_arg_0 = w3_strict_abi.codec.encode_abi(["bytes2"], arg_0)
+    encoded_arg_0 = w3_strict_abi.codec.encode(["bytes2"], arg_0)
     padded_arg_0 = encoded_arg_0.ljust(32, b"\x00")
     arg_0_topic = w3_strict_abi.keccak(padded_arg_0)
     assert arg_0_topic in log_entry["topics"]

--- a/tests/core/filtering/test_utils_functions.py
+++ b/tests/core/filtering/test_utils_functions.py
@@ -125,7 +125,7 @@ from web3._utils.filters import (
 )
 def test_match_fn_with_various_data_types(w3, data, expected, match_data_and_abi):
     abi_types, match_data = zip(*match_data_and_abi)
-    encoded_data = w3.codec.encode_abi(abi_types, data)
+    encoded_data = w3.codec.encode(abi_types, data)
     assert match_fn(w3.codec, match_data_and_abi, encoded_data) == expected
 
 
@@ -189,7 +189,7 @@ def test_match_fn_with_various_data_types_strict(
     w3_strict_abi, data, expected, match_data_and_abi
 ):
     abi_types, match_data = zip(*match_data_and_abi)
-    encoded_data = w3_strict_abi.codec.encode_abi(abi_types, data)
+    encoded_data = w3_strict_abi.codec.encode(abi_types, data)
     assert match_fn(w3_strict_abi.codec, match_data_and_abi, encoded_data) == expected
 
 
@@ -203,9 +203,9 @@ def test_match_fn_with_various_data_types_strict(
         ),
     ),
 )
-def test_encode_abi_with_wrong_types_strict(w3_strict_abi, data, abi_type):
+def test_encode_with_wrong_types_strict(w3_strict_abi, data, abi_type):
     with pytest.raises(ValueOutOfBounds):
-        w3_strict_abi.codec.encode_abi(abi_type, data)
+        w3_strict_abi.codec.encode(abi_type, data)
 
 
 def test_wrong_type_match_data(w3):
@@ -215,6 +215,6 @@ def test_wrong_type_match_data(w3):
         ("string", (50505050,)),
     )
     abi_types, match_data = zip(*match_data_and_abi)
-    encoded_data = w3.codec.encode_abi(abi_types, data)
+    encoded_data = w3.codec.encode(abi_types, data)
     with pytest.raises(ValueError):
         match_fn(w3.codec, match_data_and_abi, encoded_data)

--- a/web3/_utils/async_transactions.py
+++ b/web3/_utils/async_transactions.py
@@ -8,7 +8,7 @@ from typing import (
 )
 
 from eth_abi import (
-    encode_abi,
+    encode,
 )
 from eth_typing import (
     URI,
@@ -199,7 +199,7 @@ async def async_handle_offchain_lookup(
                 # 4-byte callback function selector
                 to_bytes_if_hex(offchain_lookup_payload["callbackFunction"]),
                 # encode the `data` from the result and the `extraData` as bytes
-                encode_abi(
+                encode(
                     ["bytes", "bytes"],
                     [
                         to_bytes_if_hex(result["data"]),

--- a/web3/_utils/contracts.py
+++ b/web3/_utils/contracts.py
@@ -218,7 +218,7 @@ def encode_abi(
         argument_types,
         arguments,
     )
-    encoded_arguments = w3.codec.encode_abi(
+    encoded_arguments = w3.codec.encode(
         argument_types,
         normalized_arguments,
     )

--- a/web3/utils/async_exception_handling.py
+++ b/web3/utils/async_exception_handling.py
@@ -4,7 +4,7 @@ from typing import (
 )
 
 from eth_abi import (
-    encode_abi,
+    encode,
 )
 from eth_typing import (
     URI,
@@ -81,7 +81,7 @@ async def async_handle_offchain_lookup(
                 # 4-byte callback function selector
                 to_bytes_if_hex(offchain_lookup_payload["callbackFunction"]),
                 # encode the `data` from the result and the `extraData` as bytes
-                encode_abi(
+                encode(
                     ["bytes", "bytes"],
                     [
                         to_bytes_if_hex(result["data"]),

--- a/web3/utils/exception_handling.py
+++ b/web3/utils/exception_handling.py
@@ -4,7 +4,7 @@ from typing import (
 )
 
 from eth_abi import (
-    encode_abi,
+    encode,
 )
 from eth_typing import (
     URI,
@@ -83,7 +83,7 @@ def handle_offchain_lookup(
                 # 4-byte callback function selector
                 to_bytes_if_hex(offchain_lookup_payload["callbackFunction"]),
                 # encode the `data` from the result and the `extraData` as bytes
-                encode_abi(
+                encode(
                     ["bytes", "bytes"],
                     [
                         to_bytes_if_hex(result["data"]),

